lib/lotus/logger.rb

```diff
@@ -0,0 +1,88 @@
+require 'logger'
+
+module Lotus
+  # Lotus default logger
+  #
+  # Implement with the same interface of ruby std ::Logger
+  # Optout the logdev parameter and always use STDOUT to output the log
+  #
+  # Lotus logger also has the app_tag concept, which used to identify
+  # which application the log come from
+  #
+  # Example Usage:
+  # #1 Logger with namespace
+  # module App
+  #   class AppLogger < Lotus::Logger
+  #   end
+  #
+  #   def foo
+  #     Applogger.new.info('foo')
+  #     #=> output: I, [2015-01-10T21:55:12.727259 #80487]  INFO -- [App] : foo
+  #   end
+  # end
+  #
+  # #2 Logger without namespace
+  # class AppLogger < Lotus::Logger
+  # end
+  #
+  # Applogger.new.info('foo')
+  # #=> output: I, [2015-01-10T21:55:12.727259 #80487]  INFO -- [Shared] : foo
+  #
+  # @since 0.1.0
+
+  class Logger < ::Logger
+
+    # Override Ruby's Logger#initialize
+    #
+    # @param logdev is default to STDOUT
+    #
+    # @since 0.2.1
+    def initialize(logdev=nil, *args)
+      super(STDOUT,*args)
+      @formatter = LotusFormatter.new.tap {|f| f.app_tag = self.app_tag}
+    end
+
+    # app_tag is the identification of current app
+    # app_tag is default to use highest namespace if current namespace
+    # if app_tag is blank, lotus use default app_tag 'shared'
+    # @param logdev is default to STDOUT
+    #
+    # @since 0.2.1
+    def app_tag
+      _app_tag
+    end
+
+    # Lotus Logger default Formatter
+    #
+    # inject app_tag to logged messages
+    #
+    # @since 0.2.1
+    class LotusFormatter < ::Logger::Formatter
+      attr_accessor :app_tag
+
+      def call(severity, time, progname, msg)
+        raise Exception.new('You must set the app_tag') unless (app_tag && !app_tag.empty?)
+        progname = "[#{@app_tag}] #{progname}"
```

Comment:
@joneslee85 IMO progname has different meaning, and if you rename progname with app_name, and use both parameter and block like below, what will happen?

```ruby
logger.info('foo') { 'bar' }
#=> output: I, [2015-01-10T21:55:12.727259 #80487]  INFO -- [App] foo : bar
```